packages/headless/src/controllers/result-list/headless-result-list.ts

```diff
@@ -51,6 +51,31 @@ export function buildResultList(
     dispatch(registerFieldsToInclude(options.fieldsToInclude));
   }
 
+  let lastFetchCompleted = 0;
+  let consecutiveFetches = 0;
+  const maxConsecutiveFetches = 5;
+  const maxDelayBetweenFetches = 200;
+
+  const triggerFetchMoreResult = () => {
+    if (engine.state.search.isLoading) {
+      return;
+    }
+
+    const delayBetweenFetches = Date.now() - lastFetchCompleted;
+    if (delayBetweenFetches < maxDelayBetweenFetches) {
+      consecutiveFetches++;
+      if (consecutiveFetches >= maxConsecutiveFetches) {
+        return engine.logger.error(
```

Comment:
Open question: Should we also throw here ?

Since we would probably want the englobing app function to stop executing entirely... ?

Reply:
I think preventing further fetches until a certain delay is reached should be enough, no?
In the JSUI, manually scrolling resets the count and allows to fetch more, it doesn't completely stop execution